client/lib/plans/personal-plan.js

```diff
@@ -32,18 +44,33 @@ export const personalPlan = {
 	available: 'yes',
 	bundle_product_ids: [ 12, 9, 50, 5, 6, 46, 54, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 72, 73, 74, 75, 16 ],
 	bill_period_label: translate( 'per year' ),
-	price: '$71.88',
-	formatted_price: '$71.88',
-	raw_price: 71.88,
 	has_domain_credit: true
 };
 
+const getCurrencyCode = plans => head( map( plans, property( 'currency_code' ) ) ) || 'USD';
+
+const applyCurrency = currencyCode => {
+	const cost = personalPlan.prices[ currencyCode ];
+	const price = formatCurrency( cost, currencyCode );
+
+	return {
+		...personalPlan,
+		cost,
+		price,
+		raw_price: cost,
+		formatted_price: price,
+		currency_code: currencyCode
+	};
+};
+
+const formatPlan = flow( getCurrencyCode, applyCurrency );
+
 export const insertPersonalPlan = plans => {
 	const freePlanIndex = findIndex( plans, matchesProperty( 'product_slug', PLAN_FREE ) );
-	const hasPersonalPlan = plans.some( matchesProperty( 'product_slug', PLAN_PERSONAL ) );
+	const hasPersonalPlan = some( plans, matchesProperty( 'product_slug', PLAN_PERSONAL ) );
 
 	return ! hasPersonalPlan
-		? [ ...plans.slice( 0, freePlanIndex + 1 ), personalPlan, ...plans.slice( freePlanIndex + 1 ) ]
+		? [ ...plans.slice( 0, freePlanIndex + 1 ), formatPlan( plans ), ...plans.slice( freePlanIndex + 1 ) ]
```

Comment:
The idea behind this is that either an array of plans, or an object with plans as keys goes in, and the personal plan with the right currency goes out, considering that the input does not have the personal plan.

Why is `getCurrencyCode` implemented in this way? This was the first version:
```js
const getCurrencyCode = plans => {
	const codes = uniq( map( plans, property( 'currency_code' ) ) );
	return size( codes ) === 1
		? head( codes )
		: 'USD'
}
```

I'm using `map` to get the same values from two different object schemas. I could have done something like this
```js
const getCurrencyCode = plans => get( { plans }, 'plans[0].currency_code' ) || 'USD';
```

but the problem is that it won't work with calls to `/sites/:siteid/plans`, and I wanted one function. `map` flattens the objects returned from `/plans` and `/sites/:siteId/plans` so I always have an array of `currency_codes`. And because I trust that all the plans have the same `currency_code`, I just take the first.
Did this clarifies things a bit? I've been overcomplicating things lately 😦 . Is it better to have two specialized functions, or one generic function?

```js
const getCurrencyCodeFromPlans = plans => get( { plans }, 'plans[0].currency_code' ) || 'USD';
const getCurrencyCodeFromSitePlans = plans => get( plans, '1.currency_code' ) || 'USD';
```